week2/homework/src/index.js

```diff
@@ -1,3 +1,73 @@
 'use strict';
 
 // TODO: Write the homework code in this file
+var program = require('commander');
+const { readFile, writeFile } = require('fs');
+const { promisify } = require('util');
+
+const readFileWithPromise = promisify(readFile);
+const writeFileWithPromise = promisify(writeFile);
+
+const [ , , cmd, ...args] = process.argv;
+
+const TODO_FILE = 'todos.json'; 
+
+async function main(){
+  const data =JSON.parse( await readFileWithPromise(TODO_FILE, 'utf8').catch(() => '[]'));
+  const elementKey = parseInt(args)-1;
+
+  switch (cmd){
+
+    case 'add':
+      data.push(args.join(' '))
+      await writeFileWithPromise(TODO_FILE, JSON.stringify(data));
+      console.info('Element has been added')
+    break;
+
+    case 'list':
+      console.info(data);
+    break;
+
+    case 'remove':
+    if(data.length >= elementKey || Number.isNaN(elementKey))
+    {
+      data.splice(elementKey, 1)
+      await writeFileWithPromise(TODO_FILE, JSON.stringify(data));
+      console.info('Element is removed')
+    }
+    else
+    {
+      console.info('This element does not exist')
+    }
+    break;
+
+    case 'reset':
+      await writeFileWithPromise(TODO_FILE, JSON.stringify([]));
+    break; 
+
+    case 'update':
+  
+      if(data.length >= elementKey)
+      {
+        args[0] = '';
```

Comment:
This is a bit odd. You have other (cleaner) ways of selecting the `args` that you actually want here... `Array.prototype.shift` for instance...